proposals/devcontainer-collections.md

```diff
@@ -0,0 +1,110 @@
+
+## Collection file (devcontainer-collection.json)
+
+- Created as a release of a repository that either contains or represents groups of definitions/features from the same user/owner/publisher. (e.g. same organization in Github.)
+- Contains metadata for each definition/feature, repository origin, download link, checksum or signing information.
+- All referenced definitions/features would belong to the same user/organization. (e.g. download links would be the same organization in GitHub.)
+- References exact releases of each definition/feature.
+
+| Property | Type | Description |
+| :--- | :--- | :--- |
+| publisher | string | The organization/user that publishes this group of definitions/features. Must match with the repository user in github.|
+| repository | string | The repository where this collection is contained.|
+| version | string | Version information of this collection list.|
+| features | array | The list of features that are contained in this collection. Same information as the metadata file for the feature.|
+| definitions | array | The list of definitions contained in this collection. Same information as the metadata file for the definition.|
+
+Creating the collections file would make some changes over the metadata of the features/definitions where they are generated.
+
+- Ids would get the collection repository added to them. (e.g. for 'https://github.com/community/features' then 'community/features' to 'myfeature1' for a feature with that id)
+- The links to the release files would be added.
+- Date.
+
+The same way creating the definitions/feature file would also be part of the release process for it so that the following metadata can be added:
+- Version.
+- Checksum.
+- Signatures.
+- Date.
+
+## Definitions/features (devcontainer-feature.json/devcontainer-definition.json)
+
+### Common Properties
+
+| Property | Type | Description |
+| :--- | :--- | :--- |
+| id | string | Id of the feature/definition. The ids should be unique in the contest of the repository where they exist. |
+| displayName | string | Name of the feature/definition |
+| shortDescription | string | Optional one sentence description |
+| description | string | Description of the feature/definition |
+| repositoryPath | string | Path to the repository and folder that contains the root of the code for the feature/definition. |
+| keywords | array | List of keywords relevant to a user that would search for this definition/feature. |
+| version | string | Versioning information for the feature/definition. |
+| files | array | List of links for the download files with checksums and optional signature information. |
+
+### Definitions
+
+| Property | Type | Description |
+| :--- | :--- | :--- |
+| type | string | singleContainer/orchestrated |
+| categories | string | The categories where it belongs |
+| architectures | string | Supported architectures |
+| baseOs | string | The operating system in which the definition is based |
```

Comment:
I'm leaning more towards keeping everything in keywords and (like you mention in one of the other comments) keeping a list of recommended and recognized keywords that developers can use. And even a list of recommended 'types' of keywords. I removed 'baseOs' and 'architectures' following this idea.

Reply:
Do we want to instead document known tags/keywords?  e.g. certainly whether arm64 is support will be of great interest to macOS users. I could see filtering on what works on the chip architecture right off the bat. A list of architecture may be one that does warrant its own property otherwise.